core/templates/pages/blog-home-page/tag-filter/tag-filter.component.spec.ts

```diff
@@ -63,40 +62,36 @@ describe('Tag Filter component', () => {
     expect(component).toBeDefined();
   });
 
-  it('should initialize tag filter and select a tag and exec search', () => {
-    spyOn(component.selectionsChange, 'emit');
-    fixture.detectChanges();
-    component.selectedTags = ['tag1', 'tag2'];
-    component.listOfDefaultTags = ['tag1', 'tag2', 'tag3', 'tag4'];
-    component.filteredTags = {
-      pipe: (param1: string[], parm2: string[]) => {
-        return {
-          subscribe(callb: () => void) {
-            callb();
-          }
-        };
-      }
-    } as Observable<string[]>;
-    component.ngOnInit();
+  it('should initialize tag filter and select a tag and exec search', fakeAsync(
+    () => {
+      spyOn(component.selectionsChange, 'emit');
+      component.selectedTags = ['tag1', 'tag2'];
+      component.listOfDefaultTags = ['tag1', 'tag2', 'tag3', 'tag4'];
 
-    expect(component.filteredTags).toBeDefined();
-    expect(component.tagFilter).toBeDefined();
-    expect(component.searchDropDownTags).toEqual(['tag3', 'tag4']);
+      fixture.detectChanges();
+      component.ngOnInit();
 
-    component.tagFilterInput = {
-      nativeElement: {
-        value: ''
-      }
-    } as ElementRef;
-    component.selectTag(({ option: { viewValue: 'tag3'}}));
+      expect(component.filteredTags).toBeDefined();
+      expect(component.tagFilter).toBeDefined();
+      expect(component.searchDropDownTags).toEqual(['tag3', 'tag4']);
 
-    expect(component.selectedTags).toEqual(['tag1', 'tag2', 'tag3']);
-    expect(component.searchDropDownTags).toEqual(['tag4']);
+      component.tagFilterInput = {
+        nativeElement: {
+          value: ''
+        }
+      } as ElementRef;
+      component.selectTag(({ option: { viewValue: 'tag3'}}));
+      tick(1600);
```

Comment:
To be clear, my main concern is that a fellow developer who's reading this code will have no idea where these numbers come from. I think you should add code comments to explain. Ideally we have a common constant which both pieces of code refer to, so that if 1500 is changed then the test gets the update automatically too.

Note also that 1501 ms is 1 millisecond extra, and not 1 second extra, so I'm not sure whether you made the change that you wanted to make.

/cc @gp201

Reply:
@seanlip For frontend tests, we usually don't have common constants for the setTimeout/debounce time that is used in both `*.spec.ts` and `*.ts` files. But we could enforce this method throughout the codebase.

We also add comments only if there are multiple `tick` calls or if the `tick` is accounting for a call not present in the file directly being tested. Since the tick is usually called right after the function call, we expect developers to be able to easily find what the `tick` call is accounting for.

In some frontend tests, we observe that the function call is not completed when the time set in the `tick` is equal to the setTimeout/debounce time, and we would have to add 1 millisecond to solve it.

Reply:
OK, yes, let's use a common constant. I imagine we can use the same debounce time everywhere.

I think any presence of "magic numbers" in the codebase needs at least an explanatory comment, though using a constant is better and that shouldn't need a comment.

Reply:
@seanlip we can't use the same denounce time for the entire codebase. Since the time is selected carefully depending on the situation.

There are 2 reasons we don't create constants like that:
- if we were to declare a const in the component file we won't be able to access it in the spec file. The only way to access it would be to define it as `this.var = ...` which is not ideal. (Note that this is possible but it is not going to be a good approach.)
- as stated above if we were to define the same value in the global constants file then we would a large number of values since each value is selected based on the situation.

Reply:
Can you explain why "the time is selected carefully depending on the situation"? What factors go into selecting the correct debounce time and what would break if it was, say, 1.5 seconds everywhere?

For the second point, how many constants are there? I think having lots of individual independent constants for things like this is a maintenance burden so I would like to dig more into why further unification is not possible.

Reply:
@seanlip  @gp201 Added a comment and a constant (specific to the blgo-home-page as ther are differect values of debounce timeout being used in different places). PTAL!